.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,50 @@
+name: SonarCloud Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+          submodules: 'recursive'
+      # For building GeoNetwork, JDK8 is necessary, but for running
+      # the SonarQube plubin, JDK11 is necessary.
+      # So, first install JDK 8, build GeoNetwork, then install JDK11
+      # and run SonarQube:
+      - name: Set up JDK 8
+        uses: actions/setup-java@v3.3.0
+        with:
+          java-version: 8
+          distribution: 'temurin'
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
```

Comment:
This step is not needed if you use the `cache: maven` property of `actions/setup-java` (https://github.com/actions/setup-java#caching-packages-dependencies)